core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -858,7 +858,7 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
           try {
             // Fetch the entry first to avoid an RPC when it's already removed.
             listing.read(classOf[LogInfo], inProgressLog)
-            if (!fs.isFile(new Path(inProgressLog))) {
+            if (!fs.getFileStatus(new Path(inProgressLog)).isFile) {
```

Comment:
```
    @Deprecated
    public boolean isFile(Path f) throws IOException {
        try {
            return this.getFileStatus(f).isFile();
        } catch (FileNotFoundException var3) {
            return false;
        }
    }
```

hmm... It seems like this would cause some semantic differences. For the original API, if the `Path` does not exist, `isFile` would return false. But now it throws a `FileNotFoundException`.

Reply:
Let's wrapper it at `SparkHadoopUtil`